### PR TITLE
CONSOLE-3190: update OperatorHub filter to use nodeArchitectures instead of GOARCH

### PR DESCRIPTION
**Pull request by rhamilto**
Thanks to @jhadvig for the server changes.

Note testing should be done using a Cluster Bot cluster with the changes in this PR as the filter does not apply when running console in development mode.

Screen Shot 2022-09-21 at 11 11 12 AM
The total count of available operators is `390 items` and `window.SERVER_FLAGS.nodeArchitectures` is `['amd64']`.

### CODE REVIEW

**Comment by rhamilto** [comment]
QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

LGTM Approver:
/assign @TheRealJon

**Comment by rhamilto** [comment]
cc: @jaypoulz, @Prashanth684

File: cmd/bridge/main.go
Comment (spadgett) [inline cmd/bridge/main.go:244]: @jhadvig Should we err if there are any empty strings (aside from the default `""`)? We might be masking a bug here if the operator ever passes one.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (rhamilto) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:51]: `window.SERVER_FLAGS.nodeArchitectures` has a value of `null` if not defined.  correct, @jhadvig?

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (rhamilto) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx]: Thanks, @spadgett.  That was orphaned from a previous revision.  Updated.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (jhadvig) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:51]: It will :+1

File: cmd/bridge/main.go
Comment (jhadvig) [reply cmd/bridge/main.go:244]: > We might be masking a bug here if the operator ever passes one.

when did that ever happen ? 🥇

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (jhadvig) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:51]: Screenshot 2022-09-22 at 16 13 36

correct

File: cmd/bridge/main.go
Comment (jhadvig) [reply cmd/bridge/main.go:244]: will update ! :)

Comment (rhamilto) [reply cmd/bridge/main.go]: Updated with latest push.  @spadgett, PTAL.

File: cmd/bridge/main.go
Comment (TheRealJon) [inline cmd/bridge/main.go:252]: `bridge.FlagFatalf` exits the application if an empty string is found in `nodeArchitectures`, so this line shouldn't be necessary.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (TheRealJon) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:84]:
```suggestion
      relevantLabels.arch.some((arch) => _.includes(targetNodeArchitecturesLabels, arch))
```

**Review by TheRealJon — COMMENTED** [review]
Just a couple of small comments

File: cmd/bridge/main.go
Comment (jhadvig) [reply cmd/bridge/main.go:252]: Right, but there is a know behaviour of the `strings.Split()` method where
`strings.Split("", ",")`
will return
`[“”]`

thats why we are removing them extra.

Comment (TheRealJon) [reply cmd/bridge/main.go:252]: So the results of string.Split are stored in `nodeArchitectures` on L244, then you loop through all of the elements in `nodeArchitectures` on L246-250 and exit if any empty string is found. I still don't understand why you would then need to have logic to remove empty strings from that same array. The previous validation logic would have already exited with an error message if any empty string existed in `nodeArchitectures`.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (rhamilto) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:84]: FWIW, I used `_.some` here because the existing code is already using Lodash.

File: cmd/bridge/main.go
Comment (jhadvig) [reply cmd/bridge/main.go:252]: I see your point and to be honest I dont thing there is a clean way how to solve it due to the issue mentioned above... then Im debating if it wont be better 
```go
	nodeArchitectures := []string{}
	if *fNodeArchitectures != "" {
		for _, str := range strings.Split(*fNodeArchitectures, ",") {
			str = strings.TrimSpace(str)
			if str == "" {
				bridge.FlagFatalf("node-architectures", "list must contain name of node architectures separated by comma")
			}
			nodeArchitectures = append(nodeArchitectures, str)
		}
	}
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:84]: I'll leave it up to you if you want to get rid of some lodash here.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (spadgett) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:51]: Sorry, I confused `targetNodeArchitectures` with `targetNodeArchitecturesLabels`

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
/hold

The latest server change changed the default value of `nodeArchitectures` from `null` to `[]`.  Need to update the frontend logic.

**Comment by rhamilto** [comment]
/hold cancel

Updated https://github.com/openshift/console/pull/12072/files#diff-f5c2166645f6d9f295216d0a8e637a7e76d633490cbf3a4b94ed3aed11c5d27dR51 to account for the default value of `nodeArchitectures` to `[]`.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Review by TheRealJon — APPROVED** [review]
/lgtm

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:43]: @TheRealJon, it turns out I do need a default value here.  Mind retagging?

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by XiyunZhao** [comment]
Hi @rhamilto, does the item on the OperatorHub page change with the SERVER_FLAGS.nodeArchitecture setting

**Comment by rhamilto** [comment]
> Hi @rhamilto, does the item on the OperatorHub page change with the SERVER_FLAGS.nodeArchitecture setting

Hi, @XiyunZhao.  Yes.  You should see different results in OperatorHub based on the values of `nodeArchitectures`.  For example, `['amd64']` vs `['s390x','arm64']`.

**Comment by XiyunZhao** [comment]
This PR has been tested on private build with private operator, all checkpoint is passed
1. User can use the Flag window.SERVER_FLAGS.nodeArchitectures to check the node arch successfully. the value will be shown like ['amd64'] or ['x86_64', 's390x', 'aarch64', 'amd64', 'arm64']
2. When the user has a cluster with node architectures type equal to ['amd64', 's390x']
-      Only the operator has the labels will be shown on the operatorHub page
       operatorframework.io/arch.amd64: supported
       operatorframework.io/arch.s390x: supported
-     The item count on Operator Hub page is correct, it only counts the number of operators that are supported on the given node 
3. If one catalog source does not have an operator that is satisfied the node Architectures type, then the CatalogSource name will not list on the filter of Source

Reference: If a ClusterServiceVersion does not include an arch label, the label " operatorframework.io/arch.amd64: supported" will be added to the operator automatically

/label qe-approved

**Pull request merged**
